pkg/apis/pipeline/v1beta1/customrun_validation_test.go

```diff
@@ -0,0 +1,357 @@
+	}, {
+		name: "missing kind in Ref",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef: &v1beta1.TaskRef{
+					APIVersion: "blah",
+					Kind:       "",
+				},
+			},
+		},
+		want: apis.ErrMissingField("spec.ref.kind"),
+	}, {
+		name: "missing apiVersion in Spec",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomSpec: &v1beta1.EmbeddedCustomRunSpec{
+					TypeMeta: runtime.TypeMeta{
+						APIVersion: "",
+					},
+				},
+			},
+		},
+		want: apis.ErrMissingField("spec.spec.apiVersion"),
+	}, {
+		name: "missing kind in Spec",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomSpec: &v1beta1.EmbeddedCustomRunSpec{
+					TypeMeta: runtime.TypeMeta{
+						APIVersion: "apiVersion",
+						Kind:       "",
+					},
+				},
+			},
+		},
+		want: apis.ErrMissingField("spec.spec.kind"),
+	}, {
+		name: "invalid statusMessage in Spec",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef: &v1beta1.TaskRef{
+					APIVersion: "blah",
+					Kind:       "blah",
+				},
+				StatusMessage: v1beta1.CustomRunSpecStatusMessage(invalidStatusMessage),
+			},
+		},
+		want: apis.ErrInvalidValue(fmt.Sprintf("statusMessage should not be set if status is not set, but it is currently set to %s", invalidStatusMessage), "statusMessage"),
+	}, {
+		name: "non-unique params",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef: &v1beta1.TaskRef{
+					APIVersion: "blah",
+					Kind:       "blah",
+				},
+				Params: []v1beta1.Param{{
+					Name:  "foo",
+					Value: *v1beta1.NewStructuredValues("foo"),
+				}, {
+					Name:  "foo",
+					Value: *v1beta1.NewStructuredValues("foo"),
+				}},
+			},
+		},
+		want: apis.ErrMultipleOneOf("spec.params[foo].name"),
+	}} {
+		t.Run(c.name, func(t *testing.T) {
+			err := c.run.Validate(context.Background())
+			if d := cmp.Diff(err.Error(), c.want.Error()); d != "" {
+				t.Error(diff.PrintWantGot(d))
+			}
+		})
+	}
+}
+
+func TestRun_Valid(t *testing.T) {
+	for _, c := range []struct {
+		name string
+		run  *v1beta1.CustomRun
+	}{{
+		name: "no params",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef: &v1beta1.TaskRef{
+					APIVersion: "blah",
+					Kind:       "blah",
+					Name:       "blah",
+				},
+			},
+		},
+	}, {
+		name: "unnamed",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef: &v1beta1.TaskRef{
+					APIVersion: "blah",
+					Kind:       "blah",
+				},
+			},
+		},
+	}, {
+		name: "unnamed",
```

Comment:
Multiple unnamed?

Reply:
agreed, unsure why there were multiple tests named `unnamed`. same problem in `run_validation_test.go`.